ruby/private/toolchain.bzl

```diff
@@ -0,0 +1,47 @@
+"Repository rule for registering Ruby interpreters"
+
+load("//ruby/private:download.bzl", _rb_download = "rb_download")
+load("//ruby/private/toolchain:repository_proxy.bzl", _rb_toolchain_repository_proxy = "rb_toolchain_repository_proxy")
+
+DEFAULT_RUBY_REPOSITORY = "rules_ruby"
+
+def rb_register_toolchains(name = DEFAULT_RUBY_REPOSITORY, version = None, register = True, **kwargs):
+    """
+    Register a Ruby toolchain and lazily download the Ruby Interpreter.
+
+    * _(For MRI on Linux and macOS)_ Installed using [ruby-build](https://github.com/rbenv/ruby-build).
+    * _(For MRI on Windows)_ Installed using [RubyInstaller](https://rubyinstaller.org).
+    * _(For JRuby on any OS)_ Downloaded and installed directly from [official website](https://www.jruby.org).
+    * _(For TruffleRuby on Linux and macOS)_ Installed using [ruby-build](https://github.com/rbenv/ruby-build).
+    * _(For "system") System Ruby is used. Please note that builds are not hermetic in this case.
```

Comment:
I would say "ruby found on the PATH" to be a bit more precise about what a "System Ruby" is